### PR TITLE
base packages

### PR DESCRIPTION
**Pull request by xcoulon**
includes packages for:
- configuration
- design
- metrics
- sentry
- token parsing
- goa tooling

also includes a directory with the `makefiles` that can be `included` in a project's own `Makefile` to avoid repetition/copy-paste of goals.

### CODE REVIEW

**Comment by sbose78** [comment]
The way we do migrations is common too, could we add that too?

File: metric/metric.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	"time"
+
+	"github.com/fabric8-services/fabric8-common/log"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	namespace = ""
+	subsystem = "service"
+)
```
Comment (nurali-techie) [inline metric/metric.go]: In prometheus, any metric is identified with its name which is formed using 'Namespace', 'Subsystem' and 'Name'.
'Name' value will be same as it provide info on what is metric is for (for ex, 'request_total')
Now to distinguish whether given metric belong to X system for Y system (like for WIT component or build component), we need to have different value for either 'Namespace' or 'Subsystem'.  I guess we can have different value for 'Subsystem' and we can take it in Recorder() [link](https://github.com/fabric8-services/fabric8-common/pull/5/files#diff-9084e94e542778fda78e394172af3f56R16) which should be used while creating metric instance.

File: metric/metric.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+		Help:      "Counter of requests received into the system.",
+	}, reqLabels)
+
+	reqDuration = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: namespace,
+		Subsystem: subsystem,
+		Name:      "request_duration_seconds",
+		Help:      "Bucketed histogram of processing time (s) of requests.",
+		Buckets:   prometheus.ExponentialBuckets(0.05, 2, 8),
+	}, reqLabels)
```
Comment (nurali-techie) [inline metric/metric.go]: `Buckets` range will be required different for different component (WIT or build), we can take `Buckets` for each metrics while calling Recorder() [link](https://github.com/fabric8-services/fabric8-common/pull/5/files#diff-9084e94e542778fda78e394172af3f56R16).

File: metric/recorder.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+	"strings"
+	"time"
+
+	"github.com/fabric8-services/fabric8-common/log"
+
+	"github.com/goadesign/goa"
+)
+
+// Recorder record prometheus metrics related to http request and response.
+func Recorder() goa.Middleware {
```
Comment (nurali-techie) [inline metric/recorder.go]: For my understanding, with this Recorder() we are exposing a goa middleware which takes care of recording four pre-defined http related metrics for the user of Recorder() goa middleware.

**Comment by xcoulon** [comment]
> The way we do migrations is common too, could we add that too?

@sbose78 yes, I plan to add that in a subsequent PR, actually.

File: metric/recorder.go (hunk 3 above)
Comment (xcoulon) [reply metric/recorder.go]: yes, that's my understanding, too. It's currently used here in WIT: https://github.com/fabric8-services/fabric8-wit/blob/master/main.go#L204

File: metric/metric.go (hunk 1 above)
Comment (xcoulon) [reply metric/metric.go]: @nurali-techie this should be fixed in https://github.com/fabric8-services/fabric8-common/pull/5/commits/ea14c04528b641a4b7908c425834d207a59d1d2b

**Pull request merged**